src/matchers/toBeAfter/index.js

```diff
@@ -1,26 +1,26 @@
-import { matcherHint, printReceived } from 'jest-matcher-utils';
-
 import predicate from './predicate';
 
-const passMessage = (received, after) => () =>
-  matcherHint('.not.toBeAfter', 'received', '') +
+const passMessage = (received, after, context) => () =>
+  context.utils.matcherHint('.not.toBeAfter', 'received', '') +
   '\n\n' +
-  `Expected date to be after ${printReceived(after)} but received:\n` +
-  `  ${printReceived(received)}`;
+  `Expected date to be after ${context.utils.printReceived(after)} but received:\n` +
+  `  ${context.utils.printReceived(received)}`;
 
-const failMessage = (received, after) => () =>
-  matcherHint('.toBeAfter', 'received', '') +
+const failMessage = (received, after, context) => () =>
+  context.utils.matcherHint('.toBeAfter', 'received', '') +
   '\n\n' +
-  `Expected date to be after ${printReceived(after)} but received:\n` +
-  `  ${printReceived(received)}`;
-
-export default {
```

Comment:
keep it inside the `export` to reduce the diff a bit?

```js
export default {
  toBeAfter(date, after) {
    // blahblah
  }
}
```

Reply:
OK. The one that's a little goofy is _toResolve/index.js_, since it's `async`. Let me know if you can think of a better syntax.

```
export default {
    toResolve: async function toResolve(promise) {
````

Reply:
```js
export default {
    async toResolve(promise) {},
};
```

this should work?

Reply:
I'd tried that, but then tests fail. I don't understand why. Here's one for example.

```
 FAIL  src/matchers/toResolve/index.test.js
  ● .toResolve › passes when passed a promise that resolved

    TypeError: Cannot read property 'utils' of undefined

      11 |     const pass = await predicate(promise);
      12 |     if (pass) {
    > 13 |       return { pass: true, message: passMessage(this.utils) };
         |                                                      ^
      14 |     }
      15 |     return { pass: false, message: failMessage(this.utils) };
      16 |   }
```

Reply:
that... sounds weird. Maybe a babel bug (i.e. async to generator transform messes up caller context)? We should land #327 and retry it, and if it's still an issue I guess we can use the longer form 👍

Reply:
Yep, it works with the latest babel stuff, so you can apply the shorthand to `async` as well 🙂